external_libraries/boost_sync/include/boost/sync/detail/throw_exception.hpp

```diff
@@ -164,11 +164,7 @@ BOOST_NOINLINE BOOST_NORETURN void throw_exception(const char* func, const char*
 
 #endif // !defined(BOOST_NO_CXX11_RVALUE_REFERENCES) && !defined(BOOST_NO_CXX11_VARIADIC_TEMPLATES)
 
-#if !defined(BOOST_EXCEPTION_DISABLE)
-#define BOOST_SYNC_DETAIL_HERE BOOST_THROW_EXCEPTION_CURRENT_FUNCTION, __FILE__, __LINE__
-#else
 #define BOOST_SYNC_DETAIL_HERE BOOST_CURRENT_FUNCTION, __FILE__, __LINE__
-#endif
```

Comment:
we don't want to directly modify 3rd party libraries, can you please remove this? this issue will have to be fixed by updating boost_sync in its entirety; i am planning to do this soon.

Reply:
hmm, after reviewing #4914, we might use this in the 3.11 branch, but in develop (i.e. for this PR) we should definitely just update boost and boost_sync and not modify 3rd party libraries ourselves.

Reply:
Sounds good. Do you want me to open the PR against that branch?

Reply:
not sure at the moment, and I don't want you to open a PR only to have it closed later if we decide on a different approach. thank you for offering though!

Reply:
Sure, for now I've split out the boost_sync change here. Feel free to cherry pick whichever commits are appropriate for each branch :)

Reply:
thank you! i will need you to revert the boost_sync changes on this branch before we can merge this PR, though. sorry if that wasn't clear! the Qt 5.15 support is really worthwhile, it's just that the other changes will take some time to figure out and i don't want to hold up good changes with that.

Reply:
Oh, I figured you would just manually cherry-pick the Qt change and wait for the other one, since you linked that commit from #4914 and I didn't want it to get lost. But it already did since I split things up...

I'm just going to open another PR for the other one so it's *somewhere* and you can reference that.